test/unix/main.ml

```diff
@@ -103,25 +141,73 @@ let test_add t =
         (Printf.sprintf "Inserted value is not present anymore: %s." k)
   | l ->
       if not (List.mem v l && List.mem v' l) then
-        Alcotest.fail (Printf.sprintf "Adding duplicate values failed.")
+        Alcotest.fail "Adding duplicate values failed."
+
+let test_find_present_lwt t =
+  Lwt_list.iter_p
+    (fun (k, v) ->
+      Lwt.return
+        ( match Index.find_all t k with
+        | [] ->
+            Alcotest.fail
+              (Printf.sprintf "Wrong insertion: %s key is missing." k)
+        | l ->
+            if not (List.mem v l) then
+              Alcotest.fail
+                (Printf.sprintf "Wrong insertion: %s value is missing." v) ))
+    !l
 
 let find_present_live () = test_find_present t
 
 let find_absent_live () = test_find_absent t
 
+let no_extra_values_added () = test_no_extra_values_added t
+
+let membership () = test_mem t
+
+let find_present_live_lwt _ () = test_find_present_lwt t
+
 let find_present_restart () =
   test_find_present (Index.v ~fresh:false ~log_size index_name)
 
 let find_absent_restart () =
   test_find_absent (Index.v ~fresh:false ~log_size index_name)
 
+let no_extra_values_added_restart () =
+  test_no_extra_values_added (Index.v ~fresh:false ~log_size index_name)
+
+let membership_restart () =
+  test_mem (Index.v ~fresh:false ~log_size index_name)
+
+let fail_restart_not_shared () =
+  let exn = Failure "IO.v: cannot reset a readonly file" in
+  Alcotest.check_raises "Index readonly cannot be fresh." exn (fun () ->
+      let _ =
+        Index.v ~shared:false ~fresh:true ~readonly:true ~log_size index_name
+      in
+      ())
+
+let fail_restart_shared () =
+  let exn = RO_Not_Allowed in
+  Alcotest.check_raises "Index readonly cannot be shared." exn (fun () ->
+      let _ = Index.v ~shared:true ~readonly:true ~log_size index_name in
```

Comment:
There is also `ignore: 'a -> unit` for this pattern, but it's a matter of personal taste.
```suggestion
     ignore @@ Index.v ~shared:true ~readonly:true ~log_size index_name
```

Reply:
I'd prefer not to have `@@` if we can use parenthesis instead :)
Otherwise I agree.